packages/sync/src/class-sender.php

```diff
@@ -186,20 +195,30 @@ protected function __construct() {
 	private function init() {
 		add_action( 'jetpack_sync_before_send_queue_sync', array( $this, 'maybe_set_user_from_token' ), 1 );
 		add_action( 'jetpack_sync_before_send_queue_sync', array( $this, 'maybe_clear_user_from_token' ), 20 );
+		add_action( 'jetpack_loaded', array( $this, 'on_jetpack_loaded' ) );
 		add_filter( 'jetpack_xmlrpc_methods', array( $this, 'register_jetpack_xmlrpc_methods' ) );
 		foreach ( Modules::get_modules() as $module ) {
 			$module->init_before_send();
 		}
 	}
 
+	/**
+	 * Runs on Jetpack being ready to load its packages.
+	 *
+	 * @param Jetpack $jetpack object.
+	 */
+	public function on_jetpack_loaded( $jetpack ) {
+		$this->connection = $jetpack->get_connection();
+	}
+
 	/**
 	 * Detect if this is a XMLRPC request with a valid signature.
 	 * If so, changes the user to the new one.
 	 *
 	 * @access public
 	 */
 	public function maybe_set_user_from_token() {
-		$verified_user = \Jetpack::connection()->verify_xml_rpc_signature();
+		$verified_user = $this->connection->verify_xml_rpc_signature();
```

Comment:
If the main reason to embed a Jetpack instance in packages is using single instances of objects like the connection manager, I think we should reconsider how we manage those instances, and maybe consider a package solely for that purpose.